app/lib/api.dart

```diff
@@ -40,6 +40,86 @@ class ApiClient {
   final String token;
   final String refreshToken;
 
+  getCurrentShares(int bikeid) async {
+    final Response<dynamic> resp;
+
+    try {
+      final client = Dio(BaseOptions(
+        headers: {
+          'Api-Key': _apiKey,
+          'Authorization': 'Bearer $token',
+        },
+        responseType: ResponseType.json,
+        receiveDataWhenStatusError: true,
+      ));
+      resp = await client.get(
+          'https://api.vanmoof-api.com/v8/getBikeSharingInvitationsForBike/$bikeid');
+    } catch (e) {
+      if (e is DioError && e.response != null) {
+        throw 'Unable to obtain shares, error: ${e.response!.data.toString()}';
+      }
+      throw 'Unable to obtain shares, check your connection.';
+    }
+
+    return resp.data['invitations'] as List<dynamic>;
+  }
+
+  shareCurrentBike(int bikeid, String email, int duration) async {
+    final Response<dynamic> resp;
+
+    try {
+      final client = Dio(BaseOptions(
+        headers: {
+          'Api-Key': _apiKey,
+          'Authorization': 'Bearer $token',
+        },
+        responseType: ResponseType.json,
+        receiveDataWhenStatusError: true,
+      ));
+      var data = {
+        'bikeId': bikeid,
+        'email': email,
+        'duration': duration,
+      };
+
+      resp = await client.post(
+          'https://api.vanmoof-api.com/v8/createBikeSharingInvitation',
+          data: data);
+    } catch (e) {
+      if (e is DioError && e.response != null) {
+        throw 'Unable to obtain shares, error: ${e.response!.data.toString()}';
+      }
+      throw 'Failed to share the bike.';
+    }
+
+    return resp.data["result"] as Map<String, dynamic>;
```

Comment:
Lets move this line to within the try so we can catch the error if `resp.data["result"]` does not exists